Controllers/TagController.cs

```diff
@@ -45,4 +45,28 @@ public IActionResult CreateTag(TagCreateDTO newTag)
         _dbContext.SaveChanges();
         return Created($"/api/tag/{createdTag.Id}", createdTag);
     }
+
+    [HttpGet("{id}")]
+    [Authorize]
+    public IActionResult GetTagById(int id)
+    {
+        return Ok(_dbContext.Tags
+        .Where(t => t.Id == id)
+        .Select(t => new TagNoNavDTO
+        {
+            Id = t.Id,
+            Name = t.Name
+        }).SingleOrDefault());
+    }
+
+    [HttpPut("{id}")]
+    [Authorize]
+    public IActionResult UpdateATag(Tag newTag, int id)
+    {
+        Tag tagToUpdate = _dbContext.Tags.FirstOrDefault(t => t.Id == id);
+
+        tagToUpdate.Name = newTag.Name;
+        _dbContext.SaveChanges();
```

Comment:
You have no checks for if that Tag currently exists after finding it. So your logic for attempting to change its value and saving changes could attempt to run on Null